crates/neon/src/types_impl/map.rs

```diff
@@ -0,0 +1,163 @@
+use std::{error, fmt, mem::MaybeUninit};
+
+use crate::{
+    context::{internal::Env, Context},
+    handle::{internal::TransparentNoCopyWrapper, root::NapiRef, Handle},
+    macro_internal::NeonResultTag,
+    object::Object,
+    result::{JsResult, NeonResult, ResultExt},
+    sys::{self, raw},
+    types::{private, JsFunction, JsObject, Value},
+};
+
+use super::{JsBoolean, JsNumber, JsUndefined};
+
+#[derive(Debug)]
+#[repr(transparent)]
+/// The type of JavaScript
+/// [`Map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map)
+/// objects.
+pub struct JsMap(raw::Local);
+
+impl JsMap {
+    pub fn new<'cx, C>(cx: &mut C) -> NeonResult<Handle<'cx, Self>>
+    where
+        C: Context<'cx>,
+    {
+        let map = cx
+            .global::<JsFunction>("Map")?
+            .construct_with(cx)
+            .apply::<JsObject, _>(cx)?;
+
+        Ok(map.downcast_or_throw(cx)?)
```

Comment:
FYI, downcasting *is* necessary. Since it's possible to override `global.Map`, we don't know that it's actually the built-in. If someone were to override it, the constructor *might* return a different type.

```js
global.Map = function Map() {
    return new Array();
};

new Map() instanceof Map
// false
```